pkg/odo/cli/service/backends.go

```diff
@@ -0,0 +1,422 @@
+	}
+
+	return nil
+}
+
+func (b *ServiceCatalogBackend) CompleteServiceDelete(o *DeleteOptions, cmd *cobra.Command, args []string) (err error) {
+	return nil
+}
+
+// This CompleteServiceCreate contains logic to complete the "odo service create" call for the case of Service Catalog backend
+func (b *ServiceCatalogBackend) CompleteServiceCreate(o *CreateOptions, cmd *cobra.Command, args []string) (err error) {
+	if len(args) == 0 && !cmd.HasFlags() {
+		o.interactive = true
+
+	}
+
+	var class scv1beta1.ClusterServiceClass
+
+	if o.interactive {
+		classesByCategory, err := o.Client.GetKubeClient().ListServiceClassesByCategory()
+		if err != nil {
+			return fmt.Errorf("unable to retrieve service classes: %v", err)
+		}
+
+		if len(classesByCategory) == 0 {
+			return fmt.Errorf("no available service classes")
+		}
+
+		class, o.ServiceType = ui.SelectClassInteractively(classesByCategory)
+
+		plans, err := o.Client.GetKubeClient().ListMatchingPlans(class)
+		if err != nil {
+			return fmt.Errorf("couldn't retrieve plans for class %s: %v", class.GetExternalName(), err)
+		}
+
+		var svcPlan scv1beta1.ClusterServicePlan
+		// if there is only one available plan, we select it
+		if len(plans) == 1 {
+			for k, v := range plans {
+				o.Plan = k
+				svcPlan = v
+			}
+			klog.V(4).Infof("Plan %s was automatically selected since it's the only one available for service %s", o.Plan, o.ServiceType)
+		} else {
+			// otherwise select the plan interactively
+			o.Plan = ui.SelectPlanNameInteractively(plans, "Which service plan should we use ")
+			svcPlan = plans[o.Plan]
+		}
+
+		o.ParametersMap = ui.EnterServicePropertiesInteractively(svcPlan)
+		o.ServiceName = ui.EnterServiceNameInteractively(o.ServiceType, "How should we name your service ", o.validateServiceName)
+		o.outputCLI = commonui.Proceed("Output the non-interactive version of the selected options")
+		o.wait = commonui.Proceed("Wait for the service to be ready")
+	} else {
+
+		o.ServiceType = args[0]
+
+		// if two args are given, first is service type and second one is service name
+		if len(args) == 2 {
+			o.ServiceName = args[1]
+		} else {
+			o.ServiceName = o.ServiceType
+		}
+
+		// we convert the param list provided in the format of key=value list
+		// to a map
+		o.ParametersMap = make(map[string]string)
+		for _, kv := range o.parameters {
+			kvSlice := strings.Split(kv, "=")
+			// key value not provided in format of key=value
+			if len(kvSlice) != 2 {
+				return errors.New("parameters not provided in key=value format")
+			}
+			o.ParametersMap[kvSlice[0]] = kvSlice[1]
+		}
+	}
+	return nil
+}
+
+func (b *ServiceCatalogBackend) ValidateServiceCreate(o *CreateOptions) (err error) {
+	// make sure the service type exists
+	classPtr, err := o.Client.GetKubeClient().GetClusterServiceClass(o.ServiceType)
+	if err != nil {
+		return errors.Wrap(err, "unable to create service because Service Catalog is not enabled in your cluster")
```

Comment:
Do we need to wrap the error here? Can't you just use errors.Error since you know why it failed to create the service?

Reply:
Thanks. Fixed. I'm always confused when to use `Wrap` and when not to. 😕

Reply:
I think if the cause of the error is unclear, it makes sense to use `errors.Wrap` because it wraps the error with a custom message, that gives out something like - `<custom error message>: <err>`.